crates/hyperswitch_domain_models/src/payment_method_data.rs

```diff
@@ -954,6 +1200,42 @@ impl From<api_models::payments::BankTransferData> for BankTransferData {
     }
 }
 
+impl From<BankTransferData> for api_models::payments::additional_info::BankTransferAdditionalData {
+    fn from(value: BankTransferData) -> Self {
+        match value {
+            BankTransferData::AchBankTransfer {} => Self::Ach {},
+            BankTransferData::SepaBankTransfer {} => Self::Sepa {},
+            BankTransferData::BacsBankTransfer {} => Self::Bacs {},
+            BankTransferData::MultibancoBankTransfer {} => Self::Multibanco {},
+            BankTransferData::PermataBankTransfer {} => Self::Permata {},
+            BankTransferData::BcaBankTransfer {} => Self::Bca {},
+            BankTransferData::BniVaBankTransfer {} => Self::BniVa {},
+            BankTransferData::BriVaBankTransfer {} => Self::BriVa {},
+            BankTransferData::CimbVaBankTransfer {} => Self::CimbVa {},
+            BankTransferData::DanamonVaBankTransfer {} => Self::DanamonVa {},
+            BankTransferData::MandiriVaBankTransfer {} => Self::MandiriVa {},
+            BankTransferData::Pix { pix_key, cpf, cnpj } => Self::Pix(Box::new(
+                api_models::payments::additional_info::PixBankTransferAdditionalData {
+                    pix_key: pix_key.map(|pix_key| {
+                        Secret::from(MaskedBankAccount::from(pix_key.peek().to_owned()))
+                    }),
```

Comment:
Can you check this? I'm wrapping these unit structs with Secret, but added the impl for converting Secret<String, _> to the required types

Reply:
No I meant instead of wrapping these types itself with Secret, you can wrap its inner value with Secret as the types names also signify they are masked.